.github/workflows/rust.yml

```diff
@@ -152,6 +152,41 @@ jobs:
       - name: Run check
         run: make check-no-std
 
+  # Check that every combination of features is working properly.
+  bench_check:
+    name: bench_check
+    needs: check
+    runs-on: buildjet-8vcpu-ubuntu-2204
+    timeout-minutes: 120
+    steps:
+      - uses: actions/checkout@v3
+      - uses: rui314/setup-mold@v1
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v2
+        with:
+          version: "23.2"
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install Rust
+        run: rustup show && rustup install nightly && rustup component add rustfmt --toolchain nightly-x86_64-unknown-linux-gnu
+      - name: Install cargo-risczero
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-risczero@0.19
+      - name: Install risc0-zkvm toolchain # Use the risc0 cargo extension to install the risc0 std library for the current toolchain
+        run: cargo risczero install
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-provider: "buildjet"
+          shared-key: cargo-check-cache
+          save-if: ${{ github.ref == 'refs/heads/nightly' }}
+          workspaces: |
+            .
+            fuzz
+      - name: cargo bench check
+        run: cargo bench --bench prover_bench --features bench
```

Comment:
Have you tried to run all benches that we have? How much time would it take?

Reply:
haha, good question. let me give that a try

Reply:
`cargo bench` at root takes 6.5 mins on local. i'll try adding it as another parallel job in CI

Reply:
prover_bench_check is for the prover cycles and bench_check is for the root folder level `cargo bench`
still lesser than cargo hack.
wdyt @citizen-stig ? should we keep regular bench as well ?
<img width="402" alt="Screen Shot 2024-01-04 at 10 41 44 PM" src="https://github.com/Sovereign-Labs/sovereign-sdk/assets/816863/5f2ddfe6-2ab7-45da-a892-aa4d3a3a2760">